app/javascript/mastodon/api_types/accounts.ts

```diff
@@ -44,4 +44,5 @@ export interface ApiAccountJSON {
   limited?: boolean;
   memorial?: boolean;
   hide_collections: boolean;
+  mute_expires_at?: string | null;
```

Comment:
This is not pretty as this should be an `ApiMutedAccountJSON`, but I guess changing that would be a huge endeavour.

We should at least add a comment here to explain where this field comes from.

Alternatively, we could rename the existing one to `BaseApiAccountJSON`, create a new `ApiMutedAccountJSON` interface  that extends the base, and have `type ApiAccountJSON = BaseApiAccountJSON | ApiMutedAccountJSON`.

This would make it clear about what is happening, and maybe some day we can change this.

Another alternative is simply to have something like:

```ts
interface ApiMutedAccountJSON extends ApiAccountJSON {
  mute_expires_at?: string | null;
}
```

Then have `createAccountFromServerJSON` accept both `ApiMutedAccountJSON` and `ApiAccountJSON` (which makes sense, as at the moment in the code the 2 API results are converted to the same Immutable object)

Reply:
Yes, at the end it will be very similar to my previous proposal with `type ApiAccountJSON`, except that you expand the type in every place